pkg/api/apierrors/kinds.go

```diff
@@ -50,3 +53,5 @@ func (n Name) WithReason(reason string) Kind {
 }
 
 var ValidationFailed = Invalid.WithReason("ValidationFailed")
+
+var FileUploadTooLarge = RequestEntityTooLarge.WithReason("FileUploadTooLarge")
```

Comment:
Name the Reason as `ResourceTooLarge`.

```go
var ResourceTooLarge = apierrors.RequestEntityTooLarge.WithReason("ResourceTooLarge")
```

Define this in `pkg/portal/appresource/errors.go`. This file does not exist yet so please create it.